source/common/upstream/cluster_manager_impl.cc

```diff
@@ -343,6 +371,68 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
   }
 }
 
+bool ClusterManagerImpl::scheduleUpdate(const Cluster& cluster, uint32_t priority, bool mergeable) {
+  const auto& update_merge_window = cluster.info()->lbConfig().update_merge_window();
+  const auto timeout = DurationUtil::durationToMilliseconds(update_merge_window);
+
+  // Find pending updates for this cluster.
+  if (updates_map_.count(cluster.info()->name()) != 1) {
```

Comment:
perf nit: You can do this in a single map lookup/add since the map should never contain null (instead of the potentially 3 map operations we have now). E.g.,
```
auto& pending_update = updates_map_[cluster.info()->name()];
if (!pending_update) {
  pending_update.reset(...)
}
```